### PR TITLE
[core modules] Add transaction fee address resolution

### PR DESCRIPTION
**Pull request by tzakian**
We need to resolve addresses for transaction fees for different currencies.
However, we can't utilize the `PaymentRouter` module for this since that has a dependency on `LibraAccount`. We therefore need a different module to setup the routing information for different currencies. 

I do need to convert the `TransactionFees` module over to using the `PaymentRouter` module, but I propose we do that in a different PR.

### CODE REVIEW

File: language/ir-testsuite/tests/transaction_fee_distribution/test_txn_fee_collection.mvir [hunk 1]
```diff
@@ -5,13 +5,17 @@
 
 import 0x0.LibraAccount;
```
Comment (sblackshear) [inline language/ir-testsuite/tests/transaction_fee_distribution/test_txn_fee_collection.mvir:6]: Do we want to kill these old IR tests now that there are Move versions?

File: language/move-lang/tests/functional/translated_ir_tests/transaction_fee_distribution/test_txn_fee_collection.move [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    // Make sure that the pot is empty to start with
+    let fee_addr = TransactionFeeAccounts::transaction_fee_address<LBR::T>();
+    let transaction_fee_pot_amount = LibraAccount::balance<LBR::T>(fee_addr);
+    Transaction::assert(transaction_fee_pot_amount == 0, 0);
+    // Do some work
+    while (x > 0) {
+        x = x - 1;
+    }
+}
+
```
Comment (sblackshear) [inline language/move-lang/tests/functional/translated_ir_tests/transaction_fee_distribution/test_txn_fee_collection.move:22]: Nit: `// check: EXECUTED` (here and elsewhere)? I'm so paranoid about this after being bitten by weird matcher behavior...

File: language/ir-testsuite/tests/transaction_fee_distribution/test_txn_fee_distribution_proper_event_address.mvir [hunk 3]
```diff
@@ -1,5 +1,5 @@
 // Make sure that the `SentPaymentEvent` for transaction fee distribution
-// shows 0xFEE as the sender.
+// shows 0xFEE__ as the sender.
```
Comment (sblackshear) [inline language/ir-testsuite/tests/transaction_fee_distribution/test_txn_fee_distribution_proper_event_address.mvir:2]: ?

Comment (tzakian) [reply language/ir-testsuite/tests/transaction_fee_distribution/test_txn_fee_distribution_proper_event_address.mvir:2]: This is saying `0xFEE` is a prefix. But I should reword the comment to make that more clear.

File: language/ir-testsuite/tests/transaction_fee_distribution/test_txn_fee_collection.mvir (hunk 1 above)
Comment (tzakian) [reply language/ir-testsuite/tests/transaction_fee_distribution/test_txn_fee_collection.mvir:6]: I'm not sure if we're killing the IR tests after translating them? I'll take a look and do whatever the convention has been so far if that sounds good.

Comment (sblackshear) [reply language/ir-testsuite/tests/transaction_fee_distribution/test_txn_fee_collection.mvir:6]: Makes sense! Let's discuss in the Monday team meeting.

**Pull request closed without merging**